No clue what went wrong here..
This patch just avoids the warning message, it would work without it aswell.

--- library/pkgIndex.tcl.in.orig	2008-04-27 23:48:07.000000000 +0200
+++ library/pkgIndex.tcl.in	2008-04-27 23:48:13.000000000 +0200
@@ -4,7 +4,7 @@
 
 package ifneeded xml::c       @VERSION@ [list load   [file join $dir @Tclxml_LIB_FILE@]]
 package ifneeded xml::tcl     @VERSION@ [list source [file join $dir xml__tcl.tcl]]
-package ifneeded sgmlparser   1.1       [list source [file join $dir sgmlparser.tcl]]
+package ifneeded sgmlparser   1.0       [list source [file join $dir sgmlparser.tcl]]
 package ifneeded xpath        1.0       [list source [file join $dir xpath.tcl]]
 package ifneeded xmldep       1.0       [list source [file join $dir xmldep.tcl]]
 
